src/ariadne.c

```diff
@@ -5002,7 +4979,10 @@ TbBool triangulate_area(unsigned char *imap, long start_x, long start_y, long en
         {
             if ((ccolour) || (not_whole_map))
             {
-                tri_set_rectangle(rect_sx, rect_sy, rect_ex, rect_ey, ccolour);
+                if (!tri_set_rectangle(rect_sx, rect_sy, rect_ex, rect_ey, ccolour))
+                    break; // Run out of triangle space
```

Comment:
Shouldn't we set a log message here?

Reply:
Log message will be printed inside of `tri_set_rectangle`

Reply:
But I am not sure about "what to do next". Just let it not happens.